File: experimental/workflow_mt/producer/rust/src/producer/implementation/emitters/triggerbeaconsemitter.rs
use super::{
    broadcast, events, hub, identification, producer::Control, protocol, unbound_pack, Context,
    EmitterError,
};
use clibri::server;
use uuid::Uuid;

pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    event: protocol::TriggerBeaconsEmitter,
    filter: hub::filter::Filter,
    context: &Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    let mut broadcasting: Vec<(Vec<Uuid>, Vec<u8>)> = vec![];
    let mut broadcast_triggerbeacons =
        events::triggerbeaconsemitter::emit::<E, C>(event, filter, context, control)
            .await
            .map_err(EmitterError::Emitting)?;
    broadcasting.push((
        broadcast_triggerbeacons.0,
        unbound_pack(&0, &mut broadcast_triggerbeacons.1)?,
    ));
    for msg in broadcasting.iter_mut() {
        broadcast::<E, C>(msg, control).await?;
    }
    Ok(())
}

File: tests/workflow/producer/typescript (copy)/src/producer/events/triggerbeaconsemitter.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/triggerbeaconsemitter";

export function emit(
    event: Protocol.TriggerBeaconsEmitter,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<Output> {
    return Promise.reject(
        new Error(`Handler for event "TriggerBeaconsEmitter" isn't implemented`)
    );
}